### ISSUE TITLE
haproxy not using new certificate

### ISSUE THREAD

**Issue by billyburly** [opnsense/plugins#1088]
Imported a new certificate because my cert expired. Swapped over the web UI to the new cert successfully. When changing out the cert in HAproxy it is not being used. It is using the old cert, until reboot the old cert will be provided, after reboot HAproxy will not start. If you press the Test config button, the following error is given.

`parsing [/usr/local/etc/haproxy.conf:94] : 'bind 0.0.0.0:443' : 'crt-list' : error processing line 1 in file '/tmp/haproxy/ssl/5a9b45014f5979.88246914.certlist' : unable to load SSL certificate file '/tmp/haproxy/ssl/5a9b10c9e65bd.pem' file does not exist.`

**Comment by zaggash** [comment]
Look like a ducp of https://github.com/opnsense/plugins/issues/1074

**Comment by fraenki** [comment]
@billyburly That's not the same issue as in #1074. What version of OPNsense are you using?

Please try to run the following command on the CLI and paste the output here:

```
/usr/local/opnsense/scripts/OPNsense/HAProxy/exportCerts.php
```

This will tell us wether the certificate is exported or not.

**Comment by billyburly** [comment]
Sorry forgot about the version. It's 18.7.9-amd64.

`root@durrnet:~ # /usr/local/opnsense/scripts/OPNsense/HAProxy/exportCerts.php
exported cert to /tmp/haproxy/ssl/5a9b10c9e65bd.pem
exported cert to /tmp/haproxy/ssl/5bcf1e1814ccb.pem
exported cert list to /tmp/haproxy/ssl/5a9b45014f5979.88246914.certlist`

`root@durrnet:~ # grep -A1 -B1 5a9b10c9e65bd /conf/config.xml
          <ssl_enabled>1</ssl_enabled>
          <ssl_certificates>5a9b10c9e65bd,5bcf1e1814ccb</ssl_certificates>
          <ssl_default_certificate>5a9b10c9e65bd</ssl_default_certificate>
          <ssl_customOptions/>
--
  <cert>
    <refid>5a9b10c9e65bd</refid>
    <descr>STAR_billdurr_com_2017_chain</descr>`

So it should be using a cert named start_billdurr_com_2018.

**Comment by fraenki** [comment]
```
          <ssl_certificates>5bcf1e1814ccb,5c2143ce2aa28</ssl_certificates>
          <ssl_default_certificate>5a9b10c9e65bd</ssl_default_certificate>
```

@billyburly There's the error... the new certificate (ID 5a9b10c9e65bd) is only selected as "Default Certificate", but it must _also_ be added to the "Certificates" list. Could you try this?

**Comment by billyburly** [comment]
That's a little odd, because it's in the cert list, and the old one was the default
![image](https://user-images.githubusercontent.com/741241/50427337-2a924100-0873-11e9-9ca6-132e8d93d2dd.png)

Thank's for the help, setting the new cert additionally as the default fixed it.

**Comment by fraenki** [comment]
> That's a little odd

Agreed. As a result of your report the next version includes multiple small adjustments to make this more user-friendly. :)

**Comment by a1ad** [comment]
I know it's an old one, but i have exact the same problem. in version: 21.1.4. I am using the Lets Encrypt plugin and the certs are renewed & verified in the trust but still HApoxy is using the old cert.. 

tried to reboot, rerun /usr/local/opnsense/scripts/OPNsense/HAProxy/exportCerts.php but & recreated the cert with no success.

**Comment by fraenki** [comment]
@a1ad Please don't hijack old issues, open a new issue instead.
